M2/Macaulay2/d/M2lib.c

```diff
@@ -143,6 +143,8 @@ char *system_getHistory(const int n)
   return NULL;
 }
 
+int system_historyLength() { return history_length; }
```

Comment:
Is it possible to directly export this variable to the top-level?

Reply:
That was my first thought, too.  But I couldn't figure it out -- if we do something like `setupvar("historyLength", toExpr(Ccode(int, "history_length")))`, then it will just be 0 since it gets assigned at startup and never updates.

Reply:
Maybe it needs to be a pointer?

Reply:
But we'd still need to dereference it when we need its value, so we'd have to call some kind of function.  I suppose we could add some variation of `setupvar` that does this, but I'm not sure if it's worth it.